features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/domain/RequestBodyFactory.kt

```diff
@@ -18,78 +19,45 @@ internal class RequestBodyFactory(
     private val compressor: BytesCompressor = BytesCompressor()
 ) {
 
-    fun create(
-        segment: MobileSegment,
-        serializedSegment: JsonObject
-    ): RequestBody {
-        // we need to add a new line at the end of each segment for being able to format it
-        // as an Array when read by the player
-        val segmentAsBinary = (serializedSegment.toString() + "\n").toByteArray()
-        return buildRequestBody(segment, segmentAsBinary)
+    fun create(serializedSegmentsPairs: List<Pair<MobileSegment, JsonObject>>): RequestBody {
+        return buildRequestBody(serializedSegmentsPairs)
```

Comment:
should we just remove `buildRequestBody` method and put its content inside `create`? Otherwise it is useless method, it just calls another one with the same parameters.

Reply:
yes...makes sense